src/dotnet/UI.Blazor/Components/Stepper/Step.cs

```diff
@@ -0,0 +1,31 @@
+namespace ActualChat.UI.Blazor.Components;
+
+public abstract class Step : ComponentBase
+{
+    public bool IsCompleted { get; private set; }
+    public Step? CurrentStep => Stepper.CurrentStep;
+
+    [CascadingParameter] public Stepper Stepper { get; set; } = null!;
+
+    protected override void OnInitialized()
+        => Stepper.AddStep(this);
+
+    protected abstract Task<bool> Validate();
+
+    protected abstract Task Save();
+
+    public void Refresh()
+        => StateHasChanged();
+
+    public async Task<bool> Complete()
+    {
+        if (IsCompleted)
+            return true;
```

Comment:
Return must follow w/ an empty line unless it's the last statement inside the { } block.

Reply:
Done